base-math/truncdddf.c

```diff
@@ -55,7 +58,19 @@ CONVERT_WRAPPER(
 	  {
 	    if (DFP_EXCEPTIONS_ENABLED)
 	      DFP_HANDLE_EXCEPTIONS (FE_UNDERFLOW|FE_INEXACT);
-	    return SIGNBIT(a) ? -0.0 : 0.0;
+
+	    switch (fegetround())
+	      {
+	        case FE_TONEAREST:
+	          return SIGNBIT(a) ? -__DBL_DENORM_MIN__ : __DBL_DENORM_MIN__;
```

Comment:
IEEE 754, section 4.3.2, states that:

> roundTowardPositive, the result shall be the format’s floating-point number (possibly +∞) closest
to and no less than the infinitely precise result

It is my understanding that the "**format's floating point number**" indeed includes the subnormal set.  So I believe your implementation is correct.